internal/security/bootstrapper/command/gate/command.go

```diff
@@ -0,0 +1,150 @@
+	// the command name for gating the stages of bootstrapping on other services for security
+	CommandName string = "gate"
+)
+
+type cmd struct {
+	cntx          context.Context
+	waitGroup     *sync.WaitGroup
+	loggingClient logger.LoggingClient
+	config        *config.ConfigurationStruct
+}
+
+// NewCommand creates a new cmd and parses through options if any
+func NewCommand(
+	ctx context.Context,
+	wg *sync.WaitGroup,
+	lc logger.LoggingClient,
+	conf *config.ConfigurationStruct,
+	args []string) (interfaces.Command, error) {
+
+	cmd := cmd{
+		loggingClient: lc,
+		config:        conf,
+		cntx:          ctx,
+		waitGroup:     wg,
+	}
+	var dummy string
+
+	flagSet := flag.NewFlagSet(CommandName, flag.ContinueOnError)
+	flagSet.StringVar(&dummy, "confdir", "", "") // handled by bootstrap; duplicated here to prevent arg parsing errors
+
+	err := flagSet.Parse(args)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse command: %s: %w", strings.Join(args, " "), err)
+	}
+
+	return &cmd, nil
+}
+
+// Execute implements Command and runs this command
+// command gate gates stages for bootstrapping EdgeX services for security
+func (c *cmd) Execute() (statusCode int, err error) {
+	c.loggingClient.Infof("Security bootstrapper running %s", CommandName)
+
+	bootstrapServer := tcp.NewTcpServer()
+	c.loggingClient.Debugf("init phase: attempts to start up the listener on bootstrap host: %s, port: %d",
+		c.config.StageGate.BootStrapper.Host, c.config.StageGate.BootStrapper.StartPort)
+
+	// in a separate go-routine so it won't block the main thread execution
+	go openGatingSemaphorePort(bootstrapServer, c.config.StageGate.BootStrapper.StartPort, c.loggingClient,
+		"Raising bootstrap semaphore for secure bootstrapping")
+
+	// wait on for others to be done: each of tcp dialers is a blocking call
+	c.loggingClient.Debug("Waiting on dependent semaphores required to raise the ready-to-run semaphore ...")
+	if err := tcp.DialTcp(
+		c.config.StageGate.Registry.Host,
+		c.config.StageGate.Registry.ReadyPort,
+		c.loggingClient); err != nil {
+		retErr := fmt.Errorf("found error while waiting for readiness of Consul at %s:%d, err: %v",
+			c.config.StageGate.Registry.Host, c.config.StageGate.Registry.ReadyPort, err)
+		return interfaces.StatusCodeExitWithError, retErr
+	}
+	c.loggingClient.Info("Consul is ready")
+
+	if err := tcp.DialTcp(
+		c.config.StageGate.KongDB.Host,
+		c.config.StageGate.KongDB.ReadyPort,
+		c.loggingClient); err != nil {
+		retErr := fmt.Errorf("found error while waiting for readiness of Postgres at %s:%d, err: %v",
+			c.config.StageGate.KongDB.Host, c.config.StageGate.KongDB.ReadyPort, err)
+		return interfaces.StatusCodeExitWithError, retErr
+	}
+	c.loggingClient.Info("Postgres is ready")
+
+	if err := tcp.DialTcp(
+		c.config.StageGate.Database.Host,
+		c.config.StageGate.Database.ReadyPort,
+		c.loggingClient); err != nil {
+		retErr := fmt.Errorf("found error while waiting for readiness of Redis at %s:%d, err: %v",
```

Comment:
Change `Redis` to `Database` here and below.

Reply:
changed